NOTICE

```diff
@@ -3,3 +3,61 @@ Copyright 2024 The Apache Software Foundation
 
 This product includes software developed at
 The Apache Software Foundation (https://www.apache.org/).
+
+--------------------------------------------------------------------------------
+
+This product includes code from Apache Hudi, which includes the following in
+its NOTICE file:
+
+  Apache Hudi
+  Copyright 2019-2020 The Apache Software Foundation
+
+  This product includes software developed at
+  The Apache Software Foundation (http://www.apache.org/).
+
+--------------------------------------------------------------------------------
+
+This product includes code from Apache Spark, which includes the following in
+its NOTICE file:
+
+  Apache Spark
+  Copyright 2014 and onwards The Apache Software Foundation.
+
+  This product includes software developed at
+  The Apache Software Foundation (http://www.apache.org/).
+
+
+  Export Control Notice
```

Comment:
I think we can omit the Export Control Notice and Codahale bits because we haven't borrowed any Spark code that relates to these parts of the NOTICE.